Implement system theme (dark mode) and fix background image on windows.

ctrleo opened the pull request:
This pr implements feature request #741

## Changes
- ~~Removed `background` (color) property from `list_background`, which allows the app to use the system default colour instead~~
- Added two options for `background-color` in `list_background`, which alternate depending on theme
- Removed the white filter over `list_background.png` and replicated the effect by lowering the image opacity to ~~70%~~ 10%
- Moved original `list_background.png` and it's GIMP editor file to `list_background_old.png` and `list_background_old.xcf`
- Recompiled UI files with new image
- ~~Created a subroutine in `shared.py` to detect if system is in dark mode or not~~
- Determines whether system is in dark mode or not using `QApplication.styleHints().colorScheme()` 
- Added `KCC_7.0.0.spec`, a file generated through `setup.py build_binary` on windows, to .gitignore

## This fixes...
- Aesthetic cluttering while in dark mode
- List Background photo hidden in WIndows
- My eyes while converting all my manga to kindle format

## Screenshots
### Windows
#### Before
![Screenshot 2024-12-29 204312](https://github.com/user-attachments/assets/67e4458f-ece9-4013-a8a7-ae51469ca1f9)
#### After
![Screenshot 2024-12-30 104623](https://github.com/user-attachments/assets/c245a6ae-5eda-45f1-a327-9abb15253d45)

#### After (in light mode)
![Screenshot 2024-12-30 104549](https://github.com/user-attachments/assets/2e175169-2072-4608-afdb-c0bc539b6a46)

### Linux (on Debian XFCE with Dracula theme)
#### Before
![linux-before](https://github.com/user-attachments/assets/10b8ce16-1500-4535-8f3c-9b5af4d5a67a)
#### After
![newnewkcc](https://github.com/user-attachments/assets/4d2fb2bf-0e95-4620-b035-a603c4c36b07)


Ty for considering my contribution! ^-^

axu2 commented:
I may be wrong, but there's probably a qt property you can read that tells you if it's dark mode?

https://www.qt.io/blog/dark-mode-on-windows-11-with-qt-6.5

ctrleo commented:
> I may be wrong, but there's probably a qt property you can read that tells you if it's dark mode? @ctrleo
> 
> https://www.qt.io/blog/dark-mode-on-windows-11-with-qt-6.5

omg thanks sm that would be way more efficient, I'll look into implementing it so that I can cut out the darkMode function

ctrleo commented:
I feel like the image blends too well with the text, so I'm gonna lower the opacity a little more

axu2 commented:
I think it might be better to just remove the background image entirely in dark mode.

ctrleo commented:
@axu2 its kinda cute tho :( I'm trying to think of a way we might be able to pop a grey tint over it during dark mode which would make the letters stand out more

axu2 commented:
Maybe make the image grayscale or black and white, maybe invert the colors, etc

ctrleo commented:
> Maybe make the image grayscale or black and white, maybe invert the colors on top, etc

thats an intuitive solution but it would be a challenge since it's difficult to change the image itself since it's compiled in with the ui files. I thought about putting a low opacity grey border around the text to help it stand out, which looks like this:
![Screenshot 2024-12-30 030607](https://github.com/user-attachments/assets/1bef7f8e-7740-4a9c-b874-63a6f0bd77c7)
![Screenshot 2024-12-30 030648](https://github.com/user-attachments/assets/2bf033d6-d81d-481f-9ea2-1c329f1e0248)
lmk what you think and i'll push the commit

axu2 commented:
Just ran your code. Here's what it looks like on macOS. I'm personally against making the logo so prominent, I like it's subtlety.

I prefer (original no changes):

![image](https://github.com/user-attachments/assets/fb7c0827-5c2a-4c2a-9205-008b003069c8)


over (your changes):

![image](https://github.com/user-attachments/assets/6475fc14-39c2-45f3-806d-cafc02ea6433)

Also, looks like the surrounding box and background changed colors.

For reference:

![image](https://github.com/user-attachments/assets/c92ba5fa-5cfe-4eaf-a978-2e79efa7870a)


![image](https://github.com/user-attachments/assets/65f127f7-0518-460e-9e82-9429681e0647)

ctrleo commented:
> Just ran your code. Here's what it looks like on macOS. I'm personally against making the logo so prominent at least on light mode, I like it's subtlety.
> 
> I prefer (original no changes):
> 
> ![image](https://private-user-images.githubusercontent.com/20757319/399208168-fb7c0827-5c2a-4c2a-9205-008b003069c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1Mjk0MTcsIm5iZiI6MTczNTUyOTExNywicGF0aCI6Ii8yMDc1NzMxOS8zOTkyMDgxNjgtZmI3YzA4MjctNWMyYS00YzJhLTkyMDUtMDA4YjAwMzA2OWM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDAzMjUxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1NTBlNjE5ZTg3YTQ2NTIxZDU4ZWRlZGE0MDQwOGUwNTlhMjdhZDA1MzlkOGFhNWY5Y2NmZGE4NmIxNTlhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.77X52AnE4CC1qdhN3NXDdow348NzNRYtc4Lhe5VO6Mo)
> 
> over (your changes):
> 
> ![image](https://private-user-images.githubusercontent.com/20757319/399208143-6475fc14-39c2-45f3-806d-cafc02ea6433.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1Mjk0MTcsIm5iZiI6MTczNTUyOTExNywicGF0aCI6Ii8yMDc1NzMxOS8zOTkyMDgxNDMtNjQ3NWZjMTQtMzljMi00NWYzLTgwNmQtY2FmYzAyZWE2NDMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDAzMjUxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5Y2I0YzhhYmVkMDBhOWY1NjdiMzk3ZjM2NjlhYTI5YTBhNzRlODg5OTFjY2NjY2RjNTExMTQyYmM3ODJlOWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OJp7BbXQrIuq3512LQ0eObf3Q_a4HqcBiGOgwbkCbyQ)
> 
> Also, looks like the surrounding box and background changed colors.
> 
> For reference:
> 
> ![image](https://private-user-images.githubusercontent.com/20757319/399208474-c92ba5fa-5cfe-4eaf-a978-2e79efa7870a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1Mjk0MTcsIm5iZiI6MTczNTUyOTExNywicGF0aCI6Ii8yMDc1NzMxOS8zOTkyMDg0NzQtYzkyYmE1ZmEtNWNmZS00ZWFmLWE5NzgtMmU3OWVmYTc4NzBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDAzMjUxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlNWIwOTFiY2M2ZmYzODU3N2M5ZDNlNWU2ODNmMDdlMGMwODQ5MzQzMTAxMjVlNWZjOTdiYmUwYmJmZjIyMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.80MgtXNSzg-N4RNnGopotDEnvMPuK1BRt1eJ2Mjnw6k)
> 
> ![image](https://private-user-images.githubusercontent.com/20757319/399208419-65f127f7-0518-460e-9e82-9429681e0647.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1Mjk0MTcsIm5iZiI6MTczNTUyOTExNywicGF0aCI6Ii8yMDc1NzMxOS8zOTkyMDg0MTktNjVmMTI3ZjctMDUxOC00NjBlLTllODItOTQyOTY4MWUwNjQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDAzMjUxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczYzVlMmM3MWFhZmYyOTc4NTc0M2EwYTQyMGU3MWNiZmQ3OTRjNzlmOGY3Y2EzMGU0MmRmNDhjZmRiNDIxYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hLefF42mcqk03ANIv_jtJ0QTp7EIdUA6-AUG5kNRBOk)

I do see it and I'm trying to implement some kind of mask for the logo while keeping a dark & light mode, but I can't help but notice that the text in the last image hasn't changed to white like it should have which is unusual since that bug hasn't come up in any of my testing.

ctrleo commented:
a little more progress
I lowered the opacity down to 30%, and I think its as close as im gonna get for subtle, heres the results:

![Screenshot 2024-12-30 034105](https://github.com/user-attachments/assets/6c12893e-10d9-45c1-b5b6-599c07f0949b)
![Screenshot 2024-12-30 034124](https://github.com/user-attachments/assets/af8e8014-da14-4f72-b219-2acd8c1e976c)

the reason the current logo looks the way it does is because of a white cast filter over the image which essentially blends it with the background, however, when system dark mode is activated, gives us this:

![Screenshot 2024-12-30 034303](https://github.com/user-attachments/assets/46adcd83-7f23-455e-910b-7ddbb1651785)

(For reference, the text does display the correct colours, but there's a bug which causes it to default to one of the two colours in a sceenshot)

ctrleo commented:
I do think that helped, here's my changes to the UI essentially finished:
![Screenshot 2024-12-30 040141](https://github.com/user-attachments/assets/4feb5675-aa1e-4306-bd74-9ef84c210d11)
![Screenshot 2024-12-30 035808](https://github.com/user-attachments/assets/31ef093e-36f8-47f9-9a86-b72b039d7495)
![Screenshot 2024-12-30 040028](https://github.com/user-attachments/assets/f7553f4f-7ee4-4432-827d-fbd20b79d597)
![Screenshot 2024-12-30 040106](https://github.com/user-attachments/assets/9c247e66-9b49-482d-ae01-9e74893ad0fd)

axu2 commented:
This is getting there, but I want it to look like the example in the QT link earlier:

![image](https://github.com/user-attachments/assets/61f113bc-9dfe-4725-81bd-e6d1b049388d)

e.g. the text box has a white background in light mode, etc. currently the text box has the same background color as the window.

I can look into this later, this command may help (or just search for string "fusion"):

`git grep -In fusion`

ctrleo commented:
@axu2 gm! I looked into fusion and found that, with WIndows, it really takes that odd little accent colour feature and runs with it, see below:

![Screenshot 2024-12-30 101248](https://github.com/user-attachments/assets/d97384ad-5241-44f3-b86d-4ff6b6cd4e3e)

Imo I feel like it just makes it look unnecessarily tacky, however, I'm gonna change the opacity of the logo again then I'll figure out how to brighten/darken the job list box compared to background.

ctrleo commented:
I brought it down to 10%, which I think gets us quite close to what it looked like in the original UI.

![Screenshot 2024-12-30 102453](https://github.com/user-attachments/assets/47e7ca97-3cc9-448c-b877-b9e3e113b024)
![Screenshot 2024-12-30 102405](https://github.com/user-attachments/assets/db0db708-1037-4b73-bd05-caf92746e517)

ctrleo commented:
> e.g. the text box has a white background in light mode, etc. currently the text box has the same background color as the window. And you could probably make the logo even more transparent.

I added back the background color property in `KCC_ui.py` and changed it depending on light or dark mode, heres my results:

![Screenshot 2024-12-30 104623](https://github.com/user-attachments/assets/751a12db-944b-4c39-9028-c00fb8fb7858)
![Screenshot 2024-12-30 104549](https://github.com/user-attachments/assets/c6883589-5e26-495b-9369-1ff144843c2a)

on light mode it doesnt make much difference on a low brightness, I also added opacity to the background-color attribute, because the background-image tends to get stuck behind it on linux

axu2 commented:
I appreciate your efforts, and it let to the temporary fix of dark mode via removing the background image entirely in KCC 7.1

But I'm just not sure if it's possible to preserve the original background color in light mode when adding a background image.


https://github.com/user-attachments/assets/06397ace-4e57-4782-8f7d-d7b88a6923a7
